python/python/lance/dataset.py

```diff
@@ -1075,6 +1091,35 @@ def create_index(
                     [ivf_centroids], ["_ivf_centroids"]
                 )
                 kwargs["ivf_centroids"] = ivf_centroids_batch
+
+            if pq_codebook is not None:
+                # User provided IVF centroids
+                if _check_for_numpy(pq_codebook) and isinstance(
+                    pq_codebook, np.ndarray
+                ):
+                    if (
+                        len(pq_codebook.shape) != 3
+                        or pq_codebook.shape[0] != num_sub_vectors
+                        or pq_codebook.shape[1] != 256
```

Comment:
Check this against to `2 ** nbits`?

Reply:
we don't have nbits in the python API. I'll mention it, but leave this hard coded for now I guess